my-app/src/App.jsx

```diff
@@ -88,24 +84,76 @@ export default class App extends React.Component {
     ));
 
   renderCart = () =>
-    this.state.cartList.map(cartItem => (
+    this.state.cartList.map((cartItem, index) => (
       <CartItem
+        remove={this.removeFromCart}
+        decrease={this.decrease}
+        increase={this.addToCart}
+        key={`${cartItem.id}-${index}`}
+        itemId={cartItem.id} // id-ul unic
         peak={cartItem.peak}
         county={cartItem.county}
         cottage={cartItem.cottage}
+        countItem={cartItem.countItem} // nr. de trips
       />
     ));
+  cleanCart = () => {
+    this.setState({ cartList: [] });
+  };
+
+  addToCart = itemId => {
+    // debugger;
+    const itemToAdd = this.state.mountains.find(item => item.id === itemId);
+    if (!itemToAdd) {
+      return undefined;
+    }
+
+    const cartUpdated = this.state.cartList.slice();
+    const cartItem = cartUpdated.find(item => item.id === itemId);
+    if (!cartItem) {
```

Comment:
poti incepe verificarile folosind cazul pozitiv :) nu stiu daca nu cumva si eu am facut la fel in exemplu :)
e mai ok if(cartItem) {...} else {...}

Reply:
Eu am analizat cazul cu new Item si apoi cazul cu count ++! Dar e mai bine cum zici tu :)